SYCL/ESIMD/api/functional/ctors/ctor_broadcast.hpp

```diff
@@ -0,0 +1,196 @@
+//===-- ctor_broadcast.hpp - Functions for tests on simd broadcast constructor
+//      definition. -------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides functions for tests on simd broadcast constructor.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "common.hpp"
+
+namespace esimd = sycl::ext::intel::experimental::esimd;
+
+namespace esimd_test::api::functional {
+
+// Descriptor class for the case of calling constructor in initializer context.
+struct initializer {
+  static std::string get_description() { return "initializer"; }
+
+  template <typename SrcT, typename DstT, int NumElems>
+  static void call_simd_ctor(DstT ref_data, SrcT *const out) {
+    const auto simd_by_init = esimd::simd<SrcT, NumElems>(ref_data);
+    simd_by_init.copy_to(out);
+  }
+};
+
+// Descriptor class for the case of calling constructor in variable declaration
+// context.
+struct var_decl {
+  static std::string get_description() { return "variable declaration"; }
+
+  template <typename SrcT, typename DstT, int NumElems>
+  static void call_simd_ctor(DstT ref_data, SrcT *const out) {
+    const esimd::simd<SrcT, NumElems> simd_by_var_decl(ref_data);
+    simd_by_var_decl.copy_to(out);
+  }
+};
+
+// Descriptor class for the case of calling constructor in rvalue in an
+// expression context.
+struct rval_in_expr {
+  static std::string get_description() { return "rvalue in an expression"; }
+
+  template <typename SrcT, typename DstT, int NumElems>
+  static void call_simd_ctor(DstT ref_data, SrcT *const out) {
+    esimd::simd<SrcT, NumElems> simd_by_rval;
+    simd_by_rval = esimd::simd<SrcT, NumElems>(ref_data);
+    simd_by_rval.copy_to(out);
+  }
+};
+
+// Descriptor class for the case of calling constructor in const reference
+// context.
+class const_ref {
+public:
+  static std::string get_description() { return "const reference"; }
+
+  template <typename SrcT, typename DstT, int NumElems>
+  static void call_simd_ctor(DstT ref_data, SrcT *const out) {
+    call_simd_by_const_ref<SrcT, NumElems>(
+        esimd::simd<SrcT, NumElems>(ref_data), out);
+  }
+
+private:
+  template <typename SrcT, typename DstT, int NumElems>
+  static void
+  call_simd_by_const_ref(const esimd::simd<SrcT, NumElems> &simd_by_const_ref,
+                         SrcT *const out) {
+    simd_by_const_ref.copy_to(out);
+  }
+};
+
+template <typename DstT, int NumElems, typename ContextT>
+class ConvCtorTestDescription : public ITestDescription {
```

Comment:
Do you intend to re-use it for converting constructor?

Reply:
I think no, so let's make decision when these tests will be merged